test/kitchen/tasks/kitchen.py

```diff
@@ -0,0 +1,156 @@
+            raise Exit(3)
+
+        ## get list of target OSes
+        if osversions.lower() == "all":
+            osversions = ".*"
+
+        osimages = load_targets(ctx, prov, osversions)
+
+        print("Chose os targets {}\n".format(osimages))
+        for osimage in osimages:
+            if testplatforms:
+                testplatforms += "|"
+            testplatforms += "{},{}".format(osimage, prov[osimage])
+    elif platlist:
+        # platform list should be in the form of driver,os,image
+        for entry in platlist:
+            driver, os, image = entry.split(",")
+            if provider and driver != provider:
+                print("Can only use one driver type per config ( {} != {} )\n".format(provider, driver))
+                raise Exit(1)
+
+            provider = driver
+            # check to see if we know this one
+            if not platforms.get(os):
+                print("Unknown OS in {}\n".format(entry))
+                raise Exit(4)
+            if not platforms[os].get(driver):
+                print("Unknown driver in {}\n".format(entry))
+                raise Exit(5)
+            if not platforms[os][driver].get(image):
+                print("Unknown image in {}\n".format(entry))
+                raise Exit(6)
+            if testplatforms:
+                testplatforms += "|"
+            testplatforms += "{},{}".format(image, platforms[os][driver][image])
+
+    
+
+
+    print("Using the following test platform(s)\n")
+    for logplat in testplatforms.split("|"):
+        print("  {}".format(logplat))
+
+    # create the kitchen.yml file
+    with open('tmpkitchen.yml', 'w') as kitchenyml:
+        # first read the correct driver
+        print("Adding driver file drivers/{}-driver.yml\n".format(provider))
+
+        with open("drivers/{}-driver.yml".format(provider), 'r') as driverfile:
+            kitchenyml.write(driverfile.read())
+
+        # read the generic contents
+        with open("test-definitions/platforms-common.yml", 'r') as commonfile:
+            kitchenyml.write(commonfile.read())
+
+        # now open the requested test files
+        for f in glob.glob("test-definitions/{}.yml".format(testfiles)):
+            if f.lower().endswith("platforms-common.yml"):
+                print("Skipping common file\n")
+            with open(f, 'r') as infile:
+                print("Adding file {}\n".format(f))
+                kitchenyml.write(infile.read())
+
+    env = {}
+    if uservars:
+        env = load_user_env(ctx, provider, uservars)
+    env['TEST_PLATFORMS'] = testplatforms
+    ctx.run("erb tmpkitchen.yml > kitchen.yml", env=env)
+
+def load_platforms(ctx, platformfile):
+    with open(platformfile, "r") as f:
+        platforms = json.load(f)
+    return platforms
+
+def load_targets(ctx, targethash, selections):
+    returnlist = []
+    commentpattern = re.compile("^comment")
+    for selection in selections.split(","):
+        selectionpattern = re.compile(selection)
```

Comment:
This regex needs to be stricter, otherwise you'll get weird edge cases when a platform string is a substring of another platform string (eg. `win2012` and `win2012r2`):
```suggestion
        selectionpattern = re.compile("^{}$".format(selection))
```